src/app/components/work-item-quick-add/work-item-quick-add.component.ts

```diff
@@ -105,7 +105,6 @@ export class WorkItemQuickAddComponent implements OnInit, OnDestroy, OnChanges,
     this.workItem.attributes = new Map<string, string | number>();
     this.workItem.relationships = new WorkItemRelations();
     this.workItem.type = 'workitems';
-    this.workItem.id = '42';
```

Comment:
@nimishamukherjee @joshuawilson 
1. If ID is not set
2. If ID is set to `null`
In above both cases, API creates a new work item and returns a valid response.